core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedScalarSubquery.java

```diff
@@ -123,7 +124,7 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
                     correlatedJoinNode.getInput(),
                     rewrittenSubquery,
                     correlatedJoinNode.getCorrelation(),
-                    producesSingleRow ? correlatedJoinNode.getType() : LEFT,
```

Comment:
This is just a accompanying change(no logic change here). This is needed because in subqueryPlanner correlatedJoinNode type is set to LEFT and now when this line is evaluated `correlatedJoinNode.getType()` will have LEFT(instead of INNER). I made this change to preserve the logic of using INNER join when subquery has 1 cardinality and LEFT when it has 0 cardinality.